File: ddpoliglot_v6_flutter/lib/models/data/Article.dart
// ignore_for_file: file_names

import './ArticlePhrase.dart';
import './ArticleActor.dart';

import 'package:json_annotation/json_annotation.dart';

part 'Article.g.dart';

@JsonSerializable()
class Article {
  int articleID;
  String? userID;
  String? name;
  String? language;
  String? languageTranslation;
  String? textHashCode;
  String? textSpeechFileName;
  String? videoFileName;
  List<ArticlePhrase> articlePhrases;
  List<ArticleActor> articleActors;

  Article(
    this.articleID,
    this.userID,
    this.name,
    this.language,
    this.languageTranslation,
    this.textHashCode,
    this.textSpeechFileName,
    this.videoFileName,
    this.articlePhrases,
    this.articleActors,
  );

  factory Article.fromJson(Map<String, dynamic> data) =>
      _$ArticleFromJson(data);

  Map<String, dynamic> toJson() => _$ArticleToJson(this);
}

File: ddPoliglotV6.Data/Models/Article.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ddPoliglotV6.Data.Models
{
    public class Article
    {
        public int ArticleID { get; set; }
        
        public Guid UserID { get; set; }
        
        [MaxLength(250)]
        public string Name { get; set; }
        
        [MaxLength(50)]
        public string Language { get; set; }
        
        [MaxLength(50)]
        public string LanguageTranslation { get; set; }
        
        [MaxLength(50)]
        public string TextHashCode { get; set; }
        
        [MaxLength(100)]
        public string TextSpeechFileName { get; set; }

        [MaxLength(100)]
        public string VideoFileName { get; set; }
        public ICollection<ArticlePhrase> ArticlePhrases { get; set; }
        
        public ICollection<ArticleActor> ArticleActors { get; set; }
    }
}
